Makefile

```diff
@@ -82,8 +79,8 @@ tidy:
 	go mod tidy
 
 .PHONY: lint
-lint: "$(MYGOBIN)/golangci-lint"
-	"$(MYGOBIN)/golangci-lint" run ./...
+lint:
+	go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.62.2 run ./...
```

Comment:
I recently had an issue where some project (not cli-utils) installed an old version of golangci-lint into my GOBIN and it was crashing. I thought I'm using the brew-managed version but then found out that I'm not.

There is no need to install it, we can just `go run` the required version and avoid any clashes with what the user has installed. I can undo this change if you prefer to keep it as-is, but `go run` is more hermetic vs `go install`.

Reply:
I think my preferred solution for this is larger than we would want in this PR, but if you want to tackle it in another PR, I'd take it. 

What we do in the kpt-config-sync repo is go install into an gitignored /.output/go/bin/ directory inside the repo using a GOPATH passed to the go install command. Then the binary can be executed directly using the $(GOBIN)/binary-name alias in the makefile. 

Either way tho, I would suggest just updating the version in this PR to minimize other changes.

Reply:
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L74-L76
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L417-L426
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L376-L378
- https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L242-L250

It's a little convoluted, but the Makefile basically adds the repo-local `GOBIN` in front of the `PATH` and also sets `GOLANGCI_LINT` to the path of the binary. So you can call the binary raw in scripts and still use `"$(GOLANGCI_LINT)"` as a make target dependency to make sure it got installed first. Then since it's not on the regular PATH, it doesn't affect any other projects.

We should probably do something similar here.

Reply:
I decided that it'd be easier to just bump the version vs refactoring things even further.